dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java

```diff
@@ -565,23 +575,75 @@ public static boolean versionIsGreaterThanCurrentlySupported(String semVerString
     }
 
+    /**
+     * Get the version from the WDL descriptor file content. If no version is found, return the default WDL version defined by the WDL spec.
+     * If the version is invalid, return Optional.empty()
+     *
+     * @param primaryDescriptorPath The absolute path of the descriptor SourceFile to get the 'version' from
+     * @param sourceFiles           A set of SourceFiles containing the primary descriptor SourceFile and any imports
+     * @return
+     */
+    public static Optional<String> getLanguageVersion(String primaryDescriptorPath, Set<SourceFile> sourceFiles) {
+        WdlBridge wdlBridge = new WdlBridge();
+
+        Optional<SourceFile> primaryDescriptor = sourceFiles.stream().filter(sourceFile -> sourceFile.getAbsolutePath().equals(primaryDescriptorPath)).findFirst();
+        if (primaryDescriptor.isEmpty()) {
+            return Optional.empty();
+        }
+
+        final String primaryDescriptorContent = primaryDescriptor.get().getContent();
+        final Map<String, String> secondaryFiles = sourceFiles.stream()
+                .filter(sourceFile -> !sourceFile.getAbsolutePath().equals(primaryDescriptorPath))
+                .collect(Collectors.toMap(SourceFile::getAbsolutePath, SourceFile::getContent));
+        wdlBridge.setSecondaryFiles((HashMap<String, String>)secondaryFiles);
+
+        File tempMainDescriptor = null;
+        try {
+            tempMainDescriptor = File.createTempFile("main", "descriptor", Files.createTempDir());
+            Files.asCharSink(tempMainDescriptor, StandardCharsets.UTF_8).write(primaryDescriptorContent);
+            // It's possible for isVersionValid to be false for a valid 'version' so we must double-check by trying to find a 'version' string in the content
+            // Ex: Cromwell doesn't support WDL 1.1 so a 'version 1.1' workflow will return false for isVersionValid
+            final boolean isVersionValid = wdlBridge.isVersionFieldValid(tempMainDescriptor.getAbsolutePath(), primaryDescriptorPath);
+            Optional<String> parsedVersionString = getSemanticVersionString(primaryDescriptorContent);
+
+            if (parsedVersionString.isEmpty()) {
+                if (isVersionValid) {
+                    // Return default version if there's no parsed 'version' field and the version is valid (no 'version' is valid)
+                    return Optional.of(DEFAULT_WDL_VERSION);
+                }
+                // If there's no parsed 'version' string and the version isn't valid, then there's likely something wrong with the version
+                return Optional.empty();
+            }
+            return parsedVersionString;
+        } catch (IOException e) {
+            LOG.error("Error creating temporary file for descriptor {}", primaryDescriptorPath);
+            throw new CustomWebApplicationException(e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        } finally {
+            FileUtils.deleteQuietly(tempMainDescriptor);
```

Comment:
This will delete the temp file, but will it delete the temp directory that was created with `createTempDir`?

Reply:
Hmm, it was only deleting the file, but I changed it to delete the temp directory instead. This should automatically delete the contents of the temp directory according to https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html#deleteQuietly(java.io.File)